server/service/src/sync/translations/purchase_order.rs

```diff
@@ -0,0 +1,427 @@
+use repository::{
+    ChangelogRow, ChangelogTableName, EqualFilter, PurchaseOrderFilter, PurchaseOrderRepository,
+    PurchaseOrderRow, PurchaseOrderStatus, StorageConnection, SyncBufferRow,
+};
+use serde::{Deserialize, Serialize};
+use util::sync_serde::{empty_str_as_option, object_fields_as_option};
+
+use crate::sync::translations::{
+    master_list::MasterListTranslation, name::NameTranslation, period::PeriodTranslation,
+    store::StoreTranslation, PullTranslateResult, PushTranslateResult, SyncTranslation,
+};
+
+#[derive(Deserialize, Serialize, Debug, PartialEq)]
+pub enum LegacyPurchaseOrderStatus {
+    New,
+    Confirmed,
+    Authorised,
+    Finalised,
+}
+
+#[allow(non_snake_case)]
+#[derive(Deserialize, Serialize, Clone, Debug)]
+pub struct PurchaseOrderOmsFields {
+    #[serde(default)]
+    pub foreign_exchange_rate: Option<f64>,
+    #[serde(default)]
+    pub shipping_method: Option<String>,
+    #[serde(default)]
+    pub sent_datetime: Option<NaiveDateTime>,
+    #[serde(default)]
+    pub contract_signed_datetime: Option<NaiveDateTime>,
+    #[serde(default)]
+    pub advance_paid_datetime: Option<NaiveDateTime>,
+    #[serde(default)]
+    pub delivered_datetime: Option<NaiveDateTime>,
+    #[serde(default)]
+    pub received_at_port_datetime: Option<NaiveDate>,
+    #[serde(default)]
+    pub expected_delivery_datetime: Option<NaiveDate>,
+    #[serde(default)]
+    pub heading_message: Option<String>,
+}
+
+#[allow(non_snake_case)]
+#[derive(Deserialize, Serialize)]
+pub struct LegacyPurchaseOrderRow {
+    #[serde(default)]
+    #[serde(rename = "name_ID")]
+    #[serde(deserialize_with = "empty_str_as_option")]
+    pub supplier_name_link_id: Option<String>,
+    #[serde(rename = "ID")]
+    pub id: String,
+    #[serde(rename = "creation_date")]
+    pub created_datetime: NaiveDateTime,
+    #[serde(default)]
+    pub target_months: Option<f64>,
+    pub status: LegacyPurchaseOrderStatus,
+    #[serde(default)]
+    #[serde(deserialize_with = "empty_str_as_option")]
+    pub comment: Option<String>,
+    #[serde(default)]
+    #[serde(rename = "currency_ID")]
+    #[serde(deserialize_with = "empty_str_as_option")]
+    pub currency_id: Option<String>,
+    #[serde(default)]
+    #[serde(deserialize_with = "empty_str_as_option")]
+    pub reference: Option<String>,
+    #[serde(default)]
+    #[serde(rename = "confirmed_date")]
+    #[serde(deserialize_with = "empty_str_as_option")]
+    pub confirmed_datetime: Option<NaiveDateTime>,
+    // assume this is user_id - though does not reference user id in OG
+    #[serde(default)]
+    #[serde(rename = "created_by")]
+    #[serde(deserialize_with = "empty_str_as_option")]
+    pub user_id: Option<String>,
+    #[serde(rename = "store_ID")]
+    pub store_id: String,
+    #[serde(default)]
+    #[serde(deserialize_with = "empty_str_as_option")]
+    pub supplier_agent: Option<String>,
+    #[serde(default)]
+    #[serde(rename = "authorizing_officer_1")]
```

Comment:
I'm thinking it's probably better update migration and use the same name rather than re-naming here?